### ISSUE TITLE
email notification about wiki change in a watched repository

### ISSUE THREAD

**Issue by kolesar-andras** [gogits/gogs#2216]
Notification from new or modified pages in watched repositories would be nice. (Sorry if duplicate or already implemented and I missed where this feature can be activated.)

**Comment by Unknwon** [comment]
Thanks your feedback!

Can you explain and use cases?

**Comment by kolesar-andras** [comment]
We use Gogs to communicate about our projects. People involved in a project have set "watch" project repository to be informed via email when a new issue is arrives. When someone creates a new wiki page or updates an existing there is no email notification.

In another discussion (#1496) nd2s said: "There also should be notifications when a ticket gets closed and things like that. The whole email-related system needs some attention." I think we need similar features.

**Comment by Unknwon** [comment]
Thanks for following up!

> We use Gogs to communicate about our projects. People involved in a project have set "watch" project repository to be informed via email when a new issue is arrives. When someone creates a new wiki page or updates an existing there is no email notification.

So you want to get email notifications when someone updated wiki?

**Comment by kolesar-andras** [comment]
Yes, exactly.

**Comment by Unknwon** [comment]
I think this would be extremely annoying...

**Comment by kolesar-andras** [comment]
That could be a configuration option on user like this:

    [ ] notify me when wiki changes in a watched repository

**Comment by Unknwon** [comment]
There is no per-repo config for users currently...

**Comment by kolesar-andras** [comment]
I meant this as a per-user configuration, similar to notification center on GitHub.

**Comment by Unknwon** [comment]
> I meant this as a per-user configuration, similar to notification center on GitHub.

This leads to receive all wiki updates from all watched repositories, which I don't think is user-desired behavior...

**Comment by kolesar-andras** [comment]
If you have many watched repositories, you won't switch it on. Default setting could be off. I would definitely like this feature set on because I have only a few repositories.

**Comment by Unknwon** [comment]
> If you have many watched repositories, you won't switch it on. Default setting could be off. I would definitely like this feature set on because I have only a few repositories.

This is not always true, I can't just make it because `I currently satisfies xxx...`, soon enough, someone else will show up and see, `this is a bug, I want to set options for individual repository`.

**Comment by kolesar-andras** [comment]
Would you merge a pull request that implements this feature as a per-user setting? Or only per-repository?

**Comment by Unknwon** [comment]
There is a feature request for a page lists all watched repositories, I think we can add an option in this page for `receive email notification for wiki`.

**Comment by kolesar-andras** [comment]
Great, thank you.

**Comment by houtanb** [comment]
What is the status on this? I'd like to receive emails for changes to a given repository's wiki, but NOT to all wikis of repositories that I'm following.

**Comment by ciufudean** [comment]
My employer has an on-premise GHE installation, so my use case is getting notified about changes to the wiki pages for the features I am currently working on. So when signed in, I would click the Watch button on a wiki page.

**Comment by arretx** [comment]
This would have to be delivered in a digest format so as not to receive multiple updates as someone is saving the wiki with typos or if they forget how to format something and save it twice in a short period of time.  If it could be sent out once per day, twice per day, once per week, once per month, etc...that would be a user choice.

**Comment by PeterParslow** [comment]
We've created a GitHub wiki on the assumption that this would be the case - because it's "normal for Wikis" AND "normal for GitHub". So I've been surprised to find that noone gets notified of changes to the wiki - so it's hard to get engagement with the wiki to take off.

**Comment by fjv** [comment]
Getting notifications for any changes on any pages in a wiki may generate a lot of notifications during development of the wiki. 

But when a wiki has matured and changes become less frequent it would be very useful for the maintainer/watcher of the wiki to get notifications.

(i am currently trying to alert the maintainer of such a wiki when I came across this thread)

**Comment by YaguraStation** [comment]
Agreed. The atom feed might help in the meanwhile: `https://github.com/<user>/<repo>/wiki.atom`

**Comment by msemp** [comment]
I haven't even looked at the details, but someone has written something to implement this desirable feature: https://stackoverflow.com/questions/24648926/watch-a-page-in-gitlab-wiki . I'll be interested to hear if anyone tries this (and whether it works, of course).

**Comment by XJDHDR** [comment]
SourceForge allows you to watch a wiki and even individual pages for changes and sends you an email when a change is made. In fact, by default it subscribes the owner of the wiki to the entire wiki and subscribes anyone that edits pages to that page.

**Comment by carl-IOHK** [comment]
Similar to SourceFourge, Zendesk allows you to watch individual articles or sections on a help centre, which is the equivalent of a GitHub wiki, the user case for this is very strong. I as a user want to know when an article or wiki changes. 

We need this feature to find out what is being updated in the wiki by other teams in our decentralized organization so that we can republish those changes in our Zendesk Help Center. We use the help centre to aggregate help content from different repos and web sites.

**Comment by vpopescu** [comment]
Similar to confluence,  a button at the top of each page, that says "notify me of changes to this page".
Also maybe a user level function to automatically watch pages you create.

**Comment by sadikkuzu** [comment]
![image](https://github.com/gogs/gogs/assets/23168063/631504d4-9a82-4fec-9663-67450bd0f8d9)

**Comment by stpaultim** [comment]
As the primary maintainer of a project, I'd like people to feel free to make changes to the Wiki, but I'd also like to know if someone has updated the wiki so that I can review the changes.

Otherwise, I have to ask people to submit changes to me for review before they update the wiki, which seems to defeat the purpose of a wiki.
